### ISSUE TITLE
Error initializing device input - how to authorize cameras?

### ISSUE THREAD

**Issue by danyork** [johnboiles/obs-mac-virtualcam#35]
THANK YOU for writing this plugin! And thank you for providing such excellent instructions! I just went through all the steps and have a working build with a virtual camera. After doing the `codesign` command mentioned in issues #29 and #4, I can see the "OBS Virtual Camera" in Zoom. That's all great!

However, my cameras are not working. When I look through the console logs, I see the message that my version of OBS is not authorized to use the two cameras I have:

```
info: Logitech webcam: Selected device 'HD Pro Webcam C920'
info: Logitech webcam: Using preset 1280x720
error: Logitech webcam: Error while initializing device input: This app is not authorized to use HD Pro Webcam C920.
error: Failed to create source 'iMac-main screen'!
info: iMac FaceTime: Selected device 'FaceTime HD Camera (Built-in)'
info: iMac FaceTime: Using preset 1280x720
error: iMac FaceTime: Error while initializing device input: This app is not authorized to use FaceTime HD Camera (Built-in).
```

Last week I upgrade to Catalina and am now on 10.15.4. I've had to go and authorize every app to use different system resources, so it makes sense to me that I might have to do this for this new OBS build.  However... I don't know *HOW* I can do this.

Anyone have any suggestions here?

Thanks in advance!

**Comment by gxalpha** [comment]
Try settings -> privacy -> camera and look if obs is checked there
If it isn't, look through all the tabs tehre and check obs everywhere you can.

Is the "normal" obs working?

**Comment by danyork** [comment]
**Perfect!** That was exactly it. The issue was that "Terminal" did not have access to the camera.

Once I went in and gave permission to "Terminal", and then restarted Terminal (and then restarted my new OBS), the cameras were visible.

<img width="315" alt="security-privacy-terminal" src="https://user-images.githubusercontent.com/41662/79872481-d5f1ce00-83b3-11ea-930d-04f81c06a1f8.png">

This was not an issue for "normal" OBS because I had already granted that permission to "OBS" back when I had started OBS for the first time post-Catalina-upgrade.

Thank you!
